File: cpp/Minimum_Removals_to_Balance_Array_3634.cpp
class Solution {
public:
    int minRemoval(vector<int>& nums, int k) {
        sort(nums.begin(), nums.end());
        int l = 0, r = 0, maxWindow = 0;
        while (r < nums.size()) {
            while (static_cast<long long>(nums[r]) > static_cast<long long>(k) * nums[l]) l++;
            maxWindow = max(maxWindow, r - l + 1);
            r++;
        }
        return nums.size() - maxWindow;
    }
};

File: python/Minimum_Removals_to_Balance_Array_3634.py
class Solution:
    def minRemoval(self, nums: List[int], k: int) -> int:
        nums.sort()
        l, r, maxWindow = 0, 0, 0
        while r < len(nums):
            while nums[r] > k * nums[l]:
                l += 1
            maxWindow = max(maxWindow, r - l + 1)
            r += 1
        return len(nums) - maxWindow
